app/controllers/comments_controller.rb

```diff
@@ -1,6 +1,6 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
-  before_action :is_an_authorized_user, only: [:create]
+  before_action :authorize_comment, only: [:edit, :update, :destroy]
```

Comment:
Can be refactored with `before_action(except: [:new, :create]) { authorize(@comment || Comment) }`